### PR TITLE
Initial move of Azure endpoints into Sashimi

### PR DESCRIPTION
**Pull request by flin-8**

### CODE REVIEW

File: source/Server.Contracts/Endpoint/AgentlessEndpointResource.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+namespace Sashimi.Server.Contracts.Endpoint
+{
+    public abstract class AgentlessEndpointResource : EndpointResource
```
Comment (slewis74) [inline source/Server.Contracts/Endpoint/AgentlessEndpointResource.cs:3]: I'm not sold on this name. There is still an agent, it's just a Worker. These are more like "DeploysViaWorkerDeploymentTarget"

Comment (flin-8) [reply source/Server.Contracts/Endpoint/AgentlessEndpointResource.cs:3]: That's just how it was before, I think we can delete this intermediate class altogether

File: source/Sashimi.AzureCloudService/SpecialVariables.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
+namespace Sashimi.AzureCloudService
+{
+    //TODO: This is duplicated from Server while we sort out a way for Sashimi to contribute special variables.
+    static class SpecialVariables
+    {
+        public static class Action
+        {
+            public static class Azure
+            {
+                public static readonly string UseBundledAzureModules = "OctopusUseBundledAzureModules";
```
[the remark this answers is not in the document]
Comment (flin-8) [reply source/Sashimi.AzureCloudService/SpecialVariables.cs:10]: This is temporary, until we get the special variable contribution done. I will clean up the file and make it internal

File: source/Server.Contracts/Endpoint/AgentlessEndpointResource.cs (hunk 1 above)
Comment (flin-8) [reply source/Server.Contracts/Endpoint/AgentlessEndpointResource.cs]: I've delete it 🎉

**Comment by flin-8** [comment]
> Could we make all of the classes in AzureCloudServices and AzureWebApp, other than the Autofac modules, private?

We can't do that yet until we pull out the resource mappings/validators/octofront usages etc. That day will come 😁

**Comment by slewis74** [comment]
> > Could we make all of the classes in AzureCloudServices and AzureWebApp, other than the Autofac modules, private?
> 
> We can't do that yet until we pull out the resource mappings/validators/octofront usages etc. That day will come 😁

Ah, looks ok then.

**Pull request merged**